### ISSUE TITLE
Redundant maven dependencies

### ISSUE THREAD

**Issue by Glavo** [McFoggy/cssfx#45]
I noticed that CssFX on maven central is strangely dependent on `org.openjdk:javafx-javafx-controls:11`, I think this should be a bug, can it be fixed?

**Comment by McFoggy** [comment]
what would you expect instead ?

`CSSFX` requires openjfx, it is exactly what is expressed by the pom dependency

```
<dependency>
    <groupId>org.openjfx</groupId>
    <artifactId>javafx-controls</artifactId>
    <version>11</version>
</dependency>
```

We could be a bit better by setting it as `provided` but I do not see much a difference here.

**Comment by Glavo** [comment]
> what would you expect instead ?
> 
> `CSSFX` requires openjfx, it is exactly what is expressed by the pom dependency
> 
> ```
> <dependency>
>     <groupId>org.openjfx</groupId>
>     <artifactId>javafx-controls</artifactId>
>     <version>11</version>
> </dependency>
> ```
> 
> We could be a bit better by setting it as `provided` but I do not see much a difference here.

Yes, but JavaFX Unlike most Java libraries, Maven Central is one of its distribution channels, but not (or not necessarily) the most common usage. 

A large number of users are using JavaFX bundled in the JDK, not through Maven. In some cases this is the only way to go because OpenJFX distributions for many platforms are missing from Maven (e.g. win-aarch64 and linux-mips64). 

In addition, jars obtained from Maven are not suitable for use in jlink, so for some tasks we have to exclude them from dependencies.

Based on this situation, the vast majority of libraries I have seen that rely on JavaFX do not declare their dependence on OpenJFX, and I also think this is a JavaFX best practice.

The strange dependency confuses me in my project, and through Gradle's dependency analysis, I found out that this is a transitive dependency introduced by CssFX. In order to remove it, the jlink plugin for Gradle also broke in my project, and it took hours to fix it without success. 

I want CssFX to remove this dependency declaration. Almost all users are getting OpenJFX in their own way, including but not limited to downloading the OpenJFX SDK, using the Gradle/Maven plugin, or installing the JDK bundled with JavaFX. I don't think it makes sense to declare such a dependency in the POM.

**Comment by McFoggy** [comment]
@Glavo when a projetc/lib has to compile against classes of a library it is 200% good practice to declare it in its dependencies and not to rely on transitive deps or on the execution runtime even if this one provides it by default.

So should we remove the dependency to `org.openjfx:javafx-controls` in the project, for sure it is a NO.

As I said, before what could be a good pratice here would be to specify this `org.openjfx:javafx-controls` dependency as `provided` and that's what I will do shortly.

For the rest, if your tooling or environment does not support `org.openjfx:javafx-controls` just exclude it in your build chain.

> Based on this situation, the vast majority of libraries I have seen that rely on JavaFX do not declare their dependence on OpenJFX, and I also think this is a JavaFX best practice.
Sorry but I strongly disagree with that ; please look at maven libs from https://www.jfx-central.com/libraries.

**Comment by McFoggy** [comment]
version `11.5.1` is released and on its way to maven central.

dependency is now `provided`

**Comment by Glavo** [comment]
> @Glavo when a projetc/lib has to compile against classes of a library it is 200% good practice to declare it in its dependencies and not to rely on transitive deps or on the execution runtime even if this one provides it by default.

@McFoggy Oh, there seems to be some misunderstanding between us. 

I am a Gradle user. I hardly ever use Maven unless I'm maintaining projects created by other people. 

In fact as I said at the beginning, I'm talking about the POM on Maven Central. Since Gradle always generates new POMs and uploads, I didn't realize that the POM you were talking about is also the build script for this project.

All I want is for it to stop declaring transitive dependencies on JavaFX, thanks for the fix.
